### PR TITLE
Add Rust Core Fstar Definitions.

### PR DESCRIPTION
**Pull request by gaetan-sbt**
Added rust core types for use by sandwich.

### CODE REVIEW

File: proof-libs/fstar/core/Core.Marker.fst [hunk 1]
```diff
@@ -17,3 +37,12 @@ class t_Copy (h: Type) = {
 instance t_Copy_all t: t_Copy t = {
   dummy_copy_field = ()
 }
+
+class t_Clone (h: Type) = {
+  dummy_clone_field: unit
+}
+
+(** we consider everything to be copyable *)
```
Comment (W95Psp) [inline proof-libs/fstar/core/Core.Marker.fst:45]: Did you mean clonable here?

**Review by W95Psp — APPROVED** [review]
Thank you Gaëtan! :grinning:
I only have one small comment!

The ocamlformat job failing on the CI is surprising since you don't touch any single ML file :thinking: Can you rebase on main? (or merge main?)

**Comment by franziskuskiefer** [comment]
@W95Psp shall this get merged?

**Comment by W95Psp** [comment]
Absolutely, let's merge it.

**Review by W95Psp — APPROVED** [review]

**Comment by W95Psp** [comment]
Arf.. monday morning I suppose, clicked the button to fast :(

**Comment by W95Psp** [comment]
Let's merge all core-related addition we've been adding from sandwich in one go then.
@maximebuyse we discussed that last week I think: we should cherry pick core-only commits from the sandwich branch of hax, let me add an issue for that

**Pull request merged**